### PR TITLE
Move pkg/api/v1/node to pkg/util/node

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind cleanup
/sig node 

**What this PR does / why we need it**:

Package `pkg/api/v1/node` only contains two utils functions that can be moved to `pkg/util/node`.
The TODO comment here also aims to do this change: https://github.com/kubernetes/kubernetes/blob/30566b990aa13922dfc8532e76bd9a682c65699d/pkg/api/v1/node/util.go#L17

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/node/node.go [hunk 1]
```diff
@@ -212,3 +226,13 @@ func preparePatchBytesforNodeStatus(nodeName types.NodeName, oldNode *v1.Node, n
 	}
 	return patchBytes, nil
 }
+
+// IsNodeReady returns true if a node is ready; false otherwise.
+func IsNodeReady(node *v1.Node) bool {
```
Comment (andrewsykim) [inline pkg/util/node/node.go]: It seems this method is only used in this e2e test so maybe just move the method as an internal function in this file?

File: pkg/util/node/node.go [hunk 2]
```diff
@@ -143,6 +143,20 @@ func GetZoneKey(node *v1.Node) string {
 	return region + ":\x00:" + failureDomain
 }
 
+// GetNodeCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetNodeCondition(status *v1.NodeStatus, conditionType v1.NodeConditionType) (int, *v1.NodeCondition) {
```
Comment (andrewsykim) [inline pkg/util/node/node.go]: This is only used by controllers it seems so maybe `pkg/controller/util/node` is a better place?

**Comment by roycaihw** [comment]
cc @cheftako

File: pkg/util/node/node.go (hunk 2 above)
Comment (danielqsj) [reply pkg/util/node/node.go]: @andrewsykim `GetNodeCondition` is also called in e2e test.
Eg:  https://github.com/kubernetes/kubernetes/pull/73656/files#diff-4efe416092e6fea48c89ed2144d7825bR126

File: pkg/util/node/node.go (hunk 1 above)
Comment (danielqsj) [reply pkg/util/node/node.go]: Good catch, thanks.
Moved it as an internal function. PTAL

File: pkg/util/node/node.go (hunk 2 above)
Comment (smarterclayton) [reply pkg/util/node/node.go]: This code is so trivial it would be better to split it into one for e2e and one for controllers.

**Comment by anarchistHH1983** [comment]
Ready

File: pkg/util/node/node.go (hunk 2 above)
Comment (danielqsj) [reply pkg/util/node/node.go]: @smarterclayton  @andrewsykim  does [this commit](https://github.com/kubernetes/kubernetes/pull/73656/commits/e6592408af95065dbfb7209c8a45a0e13db01f0b) looks good ?

File: cmd/kubeadm/app/cmd/upgrade/node.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ import (
 	"k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient"
 	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
 	dryrunutil "k8s.io/kubernetes/cmd/kubeadm/app/util/dryrun"
-	"k8s.io/kubernetes/pkg/util/node"
+	utilnode "k8s.io/kubernetes/pkg/util/node"
```
Comment (liggitt) [inline cmd/kubeadm/app/cmd/upgrade/node.go]: all the import alias changes make this a significantly larger PR, and more difficult to review. can we drop the import changes from the files that don't need them?

Comment (danielqsj) [reply cmd/kubeadm/app/cmd/upgrade/node.go]: @liggitt , thanks for your review. I drop the import commit, PTAL. :)

**Comment by danielqsj** [comment]
ping @liggitt

**Pull request merged**